eXo/Sources/Controllers/Setting/ServerEditViewController.swift

```diff
@@ -0,0 +1,168 @@
+// License along with this software; if not, write to the Free
+// Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+// 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+
+
+import UIKit
+import SVProgressHUD
+import SwiftyJSON
+
+class ServerEditViewController: UIViewController {
+
+    
+    @IBOutlet weak var deleteButton: UIButton!
+    @IBOutlet weak var deleteButtonConstraintToBottom: NSLayoutConstraint!
+    let kDeleteButtonBottomMargin:CGFloat = 20.0
+    
+    @IBOutlet weak var textView: PlaceholderTextView!
+    var server:Server!
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        self.textView.text = server.serverURL
+        textView.placeholder = NSLocalizedString("OnBoarding.Message.EnterURL", comment: "")
+        Tool.applyBorderForView(self.deleteButton)
+
+        NSNotificationCenter.defaultCenter().addObserver(self, selector: Selector("keyboardWillShow:"), name:UIKeyboardWillShowNotification, object: nil)
+        NSNotificationCenter.defaultCenter().addObserver(self, selector: Selector("keyboardWillHide:"), name:UIKeyboardDidHideNotification, object: nil)
+
+    }
+    
+    override func didReceiveMemoryWarning() {
+        super.didReceiveMemoryWarning()
+    }
+    
+    @IBAction func saveAction(sender: AnyObject) {
+        save()
+    }
+    
+    @IBAction func deleteAction(sender: AnyObject) {
+        let alertController = UIAlertController(title: NSLocalizedString("Setting.Title.DeleteServer", comment: ""), message: NSLocalizedString("Setting.Message.DeleteServer", comment: ""), preferredStyle: UIAlertControllerStyle.Alert)
+        let cancelAction = UIAlertAction(title: NSLocalizedString("Word.Cancel", comment: ""), style: UIAlertActionStyle.Cancel) { (cancelAction) -> Void in
+        }
+        alertController.addAction(cancelAction)
+        let confirmAction = UIAlertAction(title: NSLocalizedString("Word.OK", comment: ""), style: UIAlertActionStyle.Destructive) { (confirmAction) -> Void in
+            ServerManager.sharedInstance.removeServer(self.server);
+            self.navigationController?.popViewControllerAnimated(true)
+
+        }
+        alertController.addAction(confirmAction)
+        self.presentViewController(alertController, animated: false, completion: nil)
+    }
+
+    
+    func textView(textView: UITextView, shouldChangeTextInRange range: NSRange, replacementText text: String) -> Bool {
+        if  (text.characters.last == "\n" ){
+            save()
+            textView.resignFirstResponder()
+            return false
+        } else {
+            return true
+        }
+    }
+    
+    func save () {
+        //verification of URL, http is the default protocol
+        var serverURL = textView.text
+        if ( serverURL.rangeOfString("http://") == nil && serverURL.rangeOfString("https://") == nil ) {
+            serverURL = "http://" + serverURL
+        }
+        let platformInfoURL = serverURL + "/rest/platform/info"
```

Comment:
At this point, you should strip the given URL of any path elements, in order to keep only the scheme and host.
Ex:
- http://my.server.com/portal/intranet => http://my.server.com
- my.server.com -> http://my.server.com
etc